cpp/path_module/algorithm/path.cpp

```diff
@@ -4,6 +4,69 @@
 
 #include "mgp.hpp"
 
+void Path::Elements(mgp_list *args, mgp_func_context *ctx, mgp_func_result *res, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard(memory);
+  const auto arguments = mgp::List(args);
+  auto result = mgp::Result(res);
+
+  try {
+    const auto path{arguments[0].ValuePath()};
+    mgp::List split_path(path.Length() * 2 + 1);
+    for (int i = 0; i < path.Length(); ++i) {
+      split_path.Append(mgp::Value(path.GetNodeAt(i)));
+      split_path.Append(mgp::Value(path.GetRelationshipAt(i)));
+    }
+    split_path.Append(mgp::Value(path.GetNodeAt(path.Length())));
+    result.SetValue(std::move(split_path));
+
+  } catch (const std::exception &e) {
+    result.SetErrorMessage(e.what());
+  }
+}
+
+void Path::Combine(mgp_list *args, mgp_func_context *ctx, mgp_func_result *res, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard(memory);
+  const auto arguments = mgp::List(args);
+  auto result = mgp::Result(res);
+
+  try {
+    auto path1{arguments[0].ValuePath()};
+    const auto path2{arguments[1].ValuePath()};
+
+    for (int i = 0; i < path2.Length(); ++i) {
+      path1.Expand(path2.GetRelationshipAt(i));
+    }
```

Comment:
Is this check of connecting node from apoc documentation done automatically with Expand function: " combines the paths into one if the connecting node matches" ?

Reply:
Yes, it will throw an exception if it can't connect. Same happens in APOC.

Reply:
Added comment to code saying that.